Database

rkho opened the pull request:
Fixed config.js, database populates properly in mysql

Started routing to send data to db, started retrieval.

rkho committed:
Started routing to database

rkho committed:
Merge branch 'master' of https://github.com/crewapp/crewapp into database

rkho committed:
Merge branch 'master' of https://github.com/crewapp/crewapp into database

rkho committed:
Implemented database properly, started routing to DB

arianf commented:
@rkho added a new line at the end of file

rkho committed:
Added line break to the end of each js file

arianf commented on server/server.js:
```diff
@@ -1,8 +1,8 @@
 var fs = require('fs');
 var express = require('express');
 var app = express();
+var Database = require ('./database/config.js');
```
Let's keep the require in the `routes.js`

arianf commented on server/routes.js:
```diff
@@ -0,0 +1,19 @@
+var Database = require ('/database/config.js');
```
Let's follow the same format as the other require statements.

arianf commented on server/routes.js:
```diff
@@ -0,0 +1,19 @@
+module.exports = function(app){
+  // Add new user account to database
+  app.post('/addnewuser', function(req, res){
+    Database.Users.create ({
```
create will create any username, so let's check if it already exists before creating it

arianf commented on server/routes.js:
```diff
@@ -0,0 +1,19 @@
+
+  // Add new message to database
+  app.post('/addnewmessage', function(req, res){
+    Database.Messages.create ({
```
new messages should be tied to users, I don't think this will currently work

arianf commented on server/routes.js:
```diff
@@ -0,0 +1,19 @@
+  // Add new message to database
+  app.post('/addnewmessage', function(req, res){
+    Database.Messages.create ({
+      message: req.body.message,
```
comma at end of file, doesn't follow current format

arianf replied:
We should add a body parse for this to work.

arianf commented on server/server.js:
```diff
@@ -1,8 +1,8 @@
 var express = require('express');
 var app = express();
+var Database = require ('./database/config.js');
 
```
add routes into `server.js`

arianf commented on server/routes.js:
```diff
@@ -0,0 +1,19 @@
+  app.post('/addnewmessage', function(req, res){
+    Database.Messages.create ({
+      message: req.body.message,
+      // timestamp
```
This is automatically added by the orm

rkho replied:
@arianf Can you clarify what you mean here?

arianf commented:
Also for all API request, lets prepend `/api/` for all routes.

rkho committed:
Removed Users table for now - For MVP, just Messages table with Name column that allows user to write their name in at time of sending message.

Routes.js no longer requires database as dependency.
Fixed syntax problems across files.

Prepended /api/ to all routes.

arianf commented on server/database/config.js:
```diff
@@ -1,13 +1,30 @@
   message: Sequelize.TEXT,
-  timestamp: Sequelize.DATE,
+  name: Sequelize.STRING,
+  timestamp: Sequelize.DATE
```
automatically adds timestamp... this is not needed.

rkho committed:
Removed timestamp from schema.
Removed exports from routes.

rkho committed:
Returned exports in routes.js since it broke server

arianf commented on server/routes.js:
```diff
@@ -0,0 +1,18 @@
+var Database = require ('./database/config.js');
+
+module.exports = function(app){
```
Instead of doing this, use this method:

`var router = require('express').Router();`

Then when you are calling post
`router.route('/api/addnewmessage').post(function(req, res){`
at the end of the file, have 
`module.exports = router;`

arianf commented on server/routes.js:
```diff
@@ -0,0 +1,18 @@
+  // });
+
+  // Add new message to database
+  app.post('/api/addnewmessage', function(req, res){
```
change `addnewmessage` -> `message`

rkho committed:
Changed routing to clean up the wrapper in module.exports.
Added routing for get requests.

rkho committed:
Removed unnecessary comment

arianf commented on server/database/config.js:
```diff
@@ -1,13 +1,29 @@
+var Messages = orm.define('messages', {
   message: Sequelize.TEXT,
-  timestamp: Sequelize.DATE,
+  name: Sequelize.STRING,
```
comma at the end of this, we need to decided as a group @pravipati whats the verdict on objects last comma?

rkho committed:
Confirmed adding objects to db successful!

rkho committed:
Removed comma from line 15 of config.js

arianf commented:
Looks good to me :+1: nice job!
